### PR TITLE
dotCMS/core#16756 Create persona selector option component

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<dot-avatar
+    [label]="persona.name"
+    [showDot]="persona.personalized"
+    [size]="24"
+    [url]="persona.photo"
+></dot-avatar>
+<label [ngClass]="{ 'dot-persona-selector-option__personalized': persona.personalized }">{{
```
Comment (fmontes) [inline src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.html]: Semantically, this shouldn't be a `<label>`

File: src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    persona.name
+}}</label>
+<button *ngIf="persona.personalized" (click)="deletePersonalized($event)">
```
Comment (fmontes) [inline src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.html]: We have an icon button: https://github.com/dotCMS/core-web/blob/master/src/app/view/components/_common/pattern-library/pattern-library.component.html#L50

File: src/app/view/components/dot-persona-selector/dot-persona-selector.component.ts [hunk 3]
```diff
@@ -38,6 +38,8 @@ export class DotPersonaSelectorComponent implements OnInit {
                         )
                     )
                     .subscribe((personas: DotPersona[]) => {
+                        console.log('---', personas)
```
Comment (fmontes) [inline src/app/view/components/dot-persona-selector/dot-persona-selector.component.ts]: remove.

File: src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    {{ dotMessageService.get('modes.persona.personalized') }}
```
Comment (fmontes) [inline src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.html]: Create a prop like: https://github.com/dotCMS/core-web/blob/master/src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts#L348 is not good to call methods from the html template, is old code where we do it.

File: src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.scss [hunk 5]
```diff
@@ -0,0 +1,43 @@
+@import "variables";
+
+:host {
+    align-items: center;
+    cursor: pointer;
+    display: flex;
+    height: 48px;
```
Comment (hmoreras) [inline src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.scss]: `$basic-padding * 8` or `$navigation-item-height` ?

File: src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.scss [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        }
+    }
+
+    button {
+        height: 20px;
+        margin-right: $basic-padding * 2;
+
+        ::ng-deep {
+            .ui-button-text {
+                font-size: 10px;
```
Comment (hmoreras) [inline src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.scss]: `$font-size-x-small`

File: src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        fixture = DOTTestBed.createComponent(DotPersonaSelectorOptionComponent);
+        component = fixture.componentInstance;
+        de = fixture.debugElement;
+        component.persona = mockDotPersona;
+        fixture.detectChanges();
+    });
+
+    it('should have dot-avatar with right properties', () => {
+        const avatar: DebugElement = de.query(By.css('dot-avatar'));
+        expect(avatar.componentInstance.label).toBe('Global Investor');
```
Comment (hmoreras) [inline src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.spec.ts]: compare `componentInstance` against an object instead each property ?

File: src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.scss [hunk 8]
```diff
@@ -27,31 +26,18 @@
...
-            background: $white;
-            border: solid 1px var(--color-main);
-            color: var(--color-main);
+        margin-right: $basic-padding * 2;
 
-            ::ng-deep i {
-                color: var(--color-main);
+        ::ng-deep {
+            .ui-button-text {
+                font-size: 10px;
```
Comment (fmontes) [inline src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.scss]: No, leave the button with their default design.

File: src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.scss [hunk 9]
```diff
@@ -27,31 +26,18 @@
...
+                font-weight: bold;
             }
-        }
 
-        ::ng-deep dot-icon {
-            margin-left: $basic-padding;
+            .ui-button-icon-right {
```
Comment (fmontes) [inline src/app/view/components/dot-persona-selector-option/dot-persona-selector-option.component.scss]: Same here, the the position is wrong, fix it in the button css.

**Pull request merged**